app/models/service_template.rb

```diff
@@ -38,13 +52,18 @@ class ServiceTemplate < ApplicationRecord
   virtual_column   :type_display,                 :type => :string
   virtual_column   :template_valid,               :type => :boolean
   virtual_column   :template_valid_error_message, :type => :string
+  virtual_column   :prov_type_display,            :type => :string
 
   default_value_for :service_type,  'unknown'
   default_value_for(:generic_subtype) { |st| 'custom' if st.prov_type == 'generic' }
 
   virtual_has_one :custom_actions, :class_name => "Hash"
   virtual_has_one :custom_action_buttons, :class_name => "Array"
 
+  def prov_type_display
```

Comment:
@h-kataria Bigger question is that this looks like UI code that should probably live in a UI presenter in the [Classic UI](https://github.com/ManageIQ/manageiq-ui-classic) repo?  /cc @Fryguy @dclarizio